File: src/main/java/com/coding/programmers/level1/숫자문자열과영단어.java
package com.coding.programmers.level1;

import java.util.Map;

public class 숫자문자열과영단어 {
    public static final Map<String, Integer> DIGIT = Map.of(
            "zero", 0, "one", 1, "two", 2,
            "three", 3, "four", 4, "five", 5,
            "six", 6, "seven", 7, "eight", 8, "nine", 9);

    public static void main(String[] args) {
        Solution sol = new Solution();

        System.out.println(sol.solution("one4seveneight"));
    }
    static class Solution {
        public int solution(String s) {
            char[] chars = s.toCharArray();
            StringBuilder result = new StringBuilder();
            String check = "";
            for (char ch : chars) {
                check += ch;
                Integer num = DIGIT.get(check);
                if (Character.isDigit(ch) || num != null) {
                    check = "";
                    int n = num != null ? num : Character.digit(ch, 10);
                    result.append(n);
                }
            }
            return Integer.parseInt(result.toString());
        }
    }
}

File: src/main/kotlin/com/coding/programmers/implementation/숫자문자열과영단어.kt
package com.coding.programmers.implementation

fun main() {
    val result = 숫자문자열과영단어().solution("one4seveneight")
    println(result)

}
class 숫자문자열과영단어 {
    fun solution(s: String): Int {
        val sb = StringBuilder()
        val disit = mapOf(
            "zero" to 0,"one" to 1,"two" to 2,"three" to 3,"four" to 4,
            "five" to 5, "six" to 6, "seven" to 7, "eight" to 8, "nine" to 9
        )

        var numStr =""
        for ((index, ch) in s.toCharArray().withIndex()) {
            numStr += ch;

            val num = disit.getOrDefault(numStr, -1)
            if (ch.isDigit() || num != -1) {
                sb.append(if (ch.isDigit()) ch.digitToInt() else num)
                numStr = ""
            }
        }
        return sb.toString().toInt()
    }
}